### PR TITLE
Add btrfs support to the automatic volume resizer

### PR DESCRIPTION
**Pull request by Novex**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Adds support for automatically resizing a btrfs volume when the underlying PV is increased. Without this, if a PV is increased that uses the btrfs format it will fail to mount after pod restart (causing the pod to be stuck in the Init state) with the following error until it the PV and PVC are recreated:

```
  Warning  FailedMount  14s (x12 over 8m39s)  kubelet            MountVolume.MountDevice failed while expanding volume for volume "libero-data-btrfs-alpha-2-dev" : ResizeFS.Resize - resize of format btrfs is not supported for device /dev/nvme1n1 mounted at /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/vol-01c...282
```

#### Which issue(s) this PR fixes:
n/a

#### Special notes for your reviewer:
I tested this in our v1.19.7 development cluster, but it should work fine in later versions too.

#### Does this PR introduce a user-facing change?

```release-note
Persistent Volumes formatted with the btrfs filesystem will now automatically resize when expanded.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

### CODE REVIEW

**Comment by Novex** [comment]
CLA signed

File: staging/src/k8s.io/mount-utils/resizefs_linux.go [hunk 1]
```diff
@@ -84,3 +86,16 @@ func (resizefs *ResizeFs) xfsResize(deviceMountPath string) (bool, error) {
 	resizeError := fmt.Errorf("resize of device %s failed: %v. xfs_growfs output: %s", deviceMountPath, err, string(output))
 	return false, resizeError
 }
+
+func (resizefs *ResizeFs) btrfsResize(deviceMountPath string) (bool, error) {
+	args := []string{"filesystem", "resize", "max", deviceMountPath}
+	output, err := resizefs.mounter.Exec.Command("btrfs", args...).CombinedOutput()
```
Comment (gavinfish) [inline staging/src/k8s.io/mount-utils/resizefs_linux.go:92]:
```suggestion
	output, err := resizefs.exec.Command("btrfs", args...).CombinedOutput()
```
we have changed from mounter to exec here.

Comment (Novex) [reply staging/src/k8s.io/mount-utils/resizefs_linux.go]: thanks for the direction, I didn't realise more than just the file paths got refactored earlier today

**Comment by pacoxu** [comment]
/test pull-kubernetes-unit
the bot seems to hang.

**Pull request merged**